Description: Properly check errors during authentication to avoid falling into an infinite loop.
Origin: upstream, http://websvn.kde.org/?view=rev&revision=1074179
Bug: http://bugs.kde.org/show_bug.cgi?id=203715

--- a/kioslave/imap4/imapparser.cpp
+++ b/kioslave/imap4/imapparser.cpp
@@ -262,13 +262,12 @@ imapParser::clientAuthenticate ( KIO::Sl
   }
   cmd = sendCommand (CommandPtr(new imapCommand ("AUTHENTICATE", firstCommand.toLatin1())));
 
-  while ( true )
-  {
+  int pl = 0;
+  while ( pl != -1 && !cmd->isComplete () ) {
     //read the next line
-    while (parseLoop() == 0) {
+    while ( ( pl = parseLoop() ) == 0) {
       ;
     }
-    if ( cmd->isComplete() ) break;
 
     if (!continuation.isEmpty())
     {
